### PR TITLE
Don't use starboard_platform_group install target for bundled platforms

### PR DESCRIPTION
**Pull request by niranjanyardi**
b/298729481

Add modular loader dependency on starboard install target for platforms which don't have bundle test targets

### CODE REVIEW

File: starboard/raspi/shared/install_target.gni [hunk 1]
```diff
@@ -52,6 +52,13 @@ template("install_target") {
         rebase_path(outputs[0], root_build_dir),
         rebase_path(inputs[0], root_build_dir),
       ]
+
+      if (current_toolchain == starboard_toolchain &&
+          invoker.type == "executable" && sb_is_modular && !sb_is_evergreen) {
```
Comment (oxve) [inline starboard/raspi/shared/install_target.gni:57]: Why is `!sb_is_evergreen` needed here?

In `starboard/build/config/BUILDCONFIG.gn` this lived in the loader target which is only evergreen. Not sure I understand exactly why it's moving here. Then again, I didn't fully understand why this dep was added in the first place.

**Comment by oxve** [comment]
I think it looks good. I'd want a final say from @andrewsavage1 on this since he's been working more with modular builds and toolchains than I.

**Review by oxve — APPROVED** [review]
Nominally approved, want 2nd opinion.

**Review by andrewsavage1 — APPROVED** [review]
I'd change the title/desc slightly, something like "Don't use starboard_platform_group install target for bundled platforms"

**Pull request merged**